app/models/user.js

```diff
@@ -324,15 +325,18 @@ export function addModel(dbAdapter) {
     await Promise.all(promises)
   }
 
-  User.prototype.create = async function (skip_stoplist) {
+  User.prototype.create = async function (skip_stoplist, skip_password = false) {
     this.createdAt = new Date().getTime()
     this.updatedAt = new Date().getTime()
     this.screenName = this.screenName || this.username
 
     await this.validateOnCreate(skip_stoplist)
 
     const timer = monitor.timer('users.create-time')
-    await this.initPassword()
+
+    if (!skip_password) {
+      await this.initPassword()
+    }
```

Comment:
It would be better to generate a long random password than leave it empty (null). We have internal rule that the empty string in 'hashed_password' field means blocked account. Adding another special meaning value (null) will lead to overcomplication.

Reply:
Ok, I'll change that.